### PR TITLE
Feature: Stored Preferences

### PR DESCRIPTION
**Pull request by itsnaidu**
## Tasks Done
- Implemented stored preferences
  - Queries for fetching and saving stored preference 
  - Updated schema, and API with new queries
  - Updated DashXClient with stored preference calls
- Resolved issues with the Packages
  - Updated the Apollo client package version
  - Added target dependencies to fix issues with the SPM
  - Forced issues fixed in a couple of files

- `setAccountUid` method added
- `setIdentityToken` and `identify` methods updated
- Added `schema.json` into `gitignore`

## Clean up
 - Code related to `targetInstallation`

### CODE REVIEW

**Comment by itsnaidu** [comment]
We have to update the Readme:

For schema download with:
`apollo schema:download --endpoint="https://api.dashx-staging.com/graphql" Sources/DashX/schema.json`

For codegen with:
`apollo client:codegen --target=swift --namespace=DashXGql --localSchemaFile=Sources/DashX/schema.json --includes="Graphql/*.graphql" --passthroughCustomScalars Sources/DashX/API.swift`


The user must be at the root folder of the repo to run the above commands

File: Graphql/FetchStoredPreferences.graphql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+query FetchStoredPreferences($input: FetchStoredPreferencesInput!){
```
Comment (RabidFire) [inline Graphql/FetchStoredPreferences.graphql]: Add space before `{`

File: Graphql/FetchStoredPreferences.graphql [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    fetchStoredPreferences(input: $input){
```
Comment (RabidFire) [inline Graphql/FetchStoredPreferences.graphql]: Add space before `{`

File: Graphql/SaveStoredPreferences.graphql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+mutation SaveStoredPreferences($input: SaveStoredPreferencesInput!){
```
Comment (RabidFire) [inline Graphql/SaveStoredPreferences.graphql:1]: Add space before `{`

File: Sources/DashX/DashXClient.swift [hunk 4]
```diff
@@ -97,11 +93,12 @@ public class DashXClient {
     // MARK: -- identify
 
     public func identify(_ uid: String?, withOptions: NSDictionary?) throws {
+        /* MARK: - Why do we return here
         if uid != nil {
             self.uid = uid
             return
         }
-
+         */
```
Comment (RabidFire) [inline Sources/DashX/DashXClient.swift]: - Add setIdentity
- Remove `uid` as a parameter
- Pick `uid` from options["uid"] if present, or else use self.uid
- Same goes for `anonymousUid` - pick from options["anonymousUid"] if present, or else use self.anonymousUid
-

File: Sources/DashX/DashXClient.swift [hunk 5]
```diff
@@ -362,4 +359,59 @@ public class DashXClient {
...
+        )
+        
+        DashXLog.d(tag: #function, "Calling fetchStoredPreferences with \(fetchStoredPreferencesInput)")
+        
+        let fetchStoredPreferencesQuery = DashXGql.FetchStoredPreferencesQuery(input: fetchStoredPreferencesInput)
+        
+        Network.shared.apollo.fetch(query: fetchStoredPreferencesQuery) { result in
+            switch result {
+            case .success(let graphQLResult):
+              let json = graphQLResult.data?.fetchStoredPreferences
```
Comment (RabidFire) [inline Sources/DashX/DashXClient.swift]: Return the `preferenceData` within `graphQLResult.data?.fetchStoredPreferences`

File: Sources/DashX/DashXClient.swift [hunk 6]
```diff
@@ -362,4 +359,59 @@ public class DashXClient {
...
+        let saveStoredPreferencesInput = DashXGql.SaveStoredPreferencesInput(accountUid: uid, preferenceData: preferenceData)
+        
+        DashXLog.d(tag: #function, "Calling saveStoredPreferences with \(saveStoredPreferencesInput)")
+        
+        let saveStoredPreferencesMutation = DashXGql.SaveStoredPreferencesMutation(input: saveStoredPreferencesInput)
+        
+        Network.shared.apollo.perform(mutation: saveStoredPreferencesMutation) { result in
+            switch result {
+            case .success(let graphQLResult):
+              let json = graphQLResult.data?.saveStoredPreferences
```
Comment (RabidFire) [inline Sources/DashX/DashXClient.swift:407]: Return the `preferenceData` within `graphQLResult.data?.saveStoredPreferences`

**Review by RabidFire — CHANGES_REQUESTED** [review]
- Delete schema.json file. Commit.
- After comitting, then add to .gitignore so that future devs don't re-add.
- Ensure you commit the deletion before making gitignore changes, else you'll have to force remove the file using `git rm`.

**Comment by itsnaidu** [comment]
> * Delete schema.json file. Commit.
> * After comitting, then add to .gitignore so that future devs don't re-add.
> * Ensure you commit the deletion before making gitignore changes, else you'll have to force remove the file using `git rm`.

Sure

File: Sources/DashX/DashXClient.swift [hunk 7]
```diff
@@ -63,23 +63,23 @@ public class DashXClient {
...
+    
+    public func setAccountUid(with uid: String) {
+        self.uid = uid
     }
-
-    public func setIdentityToken(to token: String) {
+    
+    public func setIdentityToken(token: String) {
         ConfigInterceptor.shared.identityToken = token
     }
```
Comment (RabidFire) [inline Sources/DashX/DashXClient.swift:81]:
```
public func setIdentity(uid, token) {

}
```

File: Sources/DashX/DashXClient.swift [hunk 8]
```diff
@@ -63,23 +63,23 @@ public class DashXClient {
...
-    func setTargetEnvironment(to environment: String) {
+    public func setTargetEnvironment(to environment: String) {
         ConfigInterceptor.shared.targetEnvironment = environment
     }
-
-    func setTargetInstallation(to targetInstallation: String) {
-        ConfigInterceptor.shared.targetInstallation = targetInstallation
+    
+    public func setAccountUid(with uid: String) {
+        self.uid = uid
```
Comment (RabidFire) [inline Sources/DashX/DashXClient.swift]: Rename the private variables: https://github.com/dashxhq/dashx-browser/blob/main/src/Client.ts#L28-L32

File: Sources/DashX/DashXClient.swift [hunk 9]
```diff
@@ -353,7 +350,7 @@ public class DashXClient {
         Network.shared.apollo.fetch(query: fetchCartQuery) { result in
           switch result {
           case .success(let graphQLResult):
-            let json = graphQLResult.data?.fetchCart
+              let json = graphQLResult.data?.fetchCart
```
Comment (RabidFire) [inline Sources/DashX/DashXClient.swift]: De-indent

File: Sources/DashX/DashXClient.swift (hunk 6 above)
Comment (RabidFire) [reply Sources/DashX/DashXClient.swift:407]: Comment not addressed

File: Sources/DashX/DashXClient.swift (hunk 7 above)
Comment (RabidFire) [reply Sources/DashX/DashXClient.swift:81]: https://github.com/dashxhq/dashx-browser/blob/main/src/Client.ts#L117-L120

**Pull request merged**